### DISCUSSION TITLE
Problem with implementation knpPaginator

### DISCUSSION THREAD

**Discussion by n3ill19** [symfony/symfony#46701] [Q&A] [answered]
|    Q                      |   A
|------------------- | ------
| Bundle version    | x.y.z 
| Symfony version | 6.0.4
| PHP version        | 8.1.6

### Support Question

Hi,

I am asking for help in creating the pagination for the index.  I've tried many ways and everyone has failed.
Below I enclose the data for the project:

![pagination_sc](https://user-images.githubusercontent.com/9141325/174262818-a1c04ea0-60ae-478d-ac6f-513c84694240.jpg)

![pagination_program](https://user-images.githubusercontent.com/9141325/174263063-37f704df-32fa-4275-a32b-20395a2871a6.jpg)

**What I must change for it to work?**

![use_pag](https://user-images.githubusercontent.com/9141325/174272589-432ed0c2-c9cc-46d7-8554-f7dfed5f728d.jpg)
![error](https://user-images.githubusercontent.com/9141325/174272805-b0f4792b-e5d6-4bb2-b1ec-3007e4fa112e.jpg)

**Comment by norkunas** [comment]
Use `SELECT .. FROM App\Entity\Movie` because short aliases were deprecated and later removed

**Reply by n3ill19** [reply]
Thanks but still don't work

**Comment by n3ill19** [comment]
<html>
<body>


[2/2] QueryException
--
Doctrine\ORM\Query\QueryException: [Syntax Error] line 0, col 7: Error: Expected IdentificationVariable \| ScalarExpression \| AggregateExpression \| FunctionDeclaration \| PartialObjectExpression \| "(" Subselect ")" \| CaseExpression, got '.'    at C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\QueryException.php:36   at Doctrine\ORM\Query\QueryException::syntaxError('line 0, col 7: Error: Expected IdentificationVariable \| ScalarExpression \| AggregateExpression \| FunctionDeclaration \| PartialObjectExpression \| "(" Subselect ")" \| CaseExpression, got \'.\'', object(QueryException))      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:494)   at Doctrine\ORM\Query\Parser->syntaxError('IdentificationVariable \| ScalarExpression \| AggregateExpression \| FunctionDeclaration \| PartialObjectExpression \| "(" Subselect ")" \| CaseExpression', array('value' => '.', 'type' => 10, 'position' => 7))      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:2308)   at Doctrine\ORM\Query\Parser->SelectExpression()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:1227)   at Doctrine\ORM\Query\Parser->SelectClause()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:914)   at Doctrine\ORM\Query\Parser->SelectStatement()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:883)   at Doctrine\ORM\Query\Parser->QueryLanguage()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:293)   at Doctrine\ORM\Query\Parser->getAST()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:393)   at Doctrine\ORM\Query\Parser->parse()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Pagination\Paginator.php:252)   at Doctrine\ORM\Tools\Pagination\Paginator->unbindUnusedQueryParams(object(Query))      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Pagination\Paginator.php:241)   at Doctrine\ORM\Tools\Pagination\Paginator->getCountQuery()      (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Pagination\Paginator.php:115)   at Doctrine\ORM\Tools\Pagination\Paginator->count()   at count(object(Paginator))      (C:\xampp\htdocs\movies\vendor\knplabs\knp-components\src\Knp\Component\Pager\Event\Subscriber\Paginate\Doctrine\ORM\QuerySubscriber.php:46)   at Knp\Component\Pager\Event\Subscriber\Paginate\Doctrine\ORM\QuerySubscriber->items(object(ItemsEvent), 'knp_pager.items', object(TraceableEventDispatcher))      (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\Debug\WrappedListener.php:111)   at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(ItemsEvent), 'knp_pager.items', object(TraceableEventDispatcher))      (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\EventDispatcher.php:230)   at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'knp_pager.items', object(ItemsEvent))      (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\EventDispatcher.php:59)   at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(ItemsEvent), 'knp_pager.items')      (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\Debug\TraceableEventDispatcher.php:152)   at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(ItemsEvent), 'knp_pager.items')      (C:\xampp\htdocs\movies\vendor\knplabs\knp-components\src\Knp\Component\Pager\Paginator.php:87)   at Knp\Component\Pager\Paginator->paginate(object(Query), 1, 3, array('pageParameterName' => 'page', 'sortFieldParameterName' => 'sort', 'sortDirectionParameterName' => 'direction', 'filterFieldParameterName' => 'filterField', 'filterValueParameterName' => 'filterValue', 'distinct' => true, 'pageOutOfRange' => 'ignore', 'defaultLimit' => 10))      (C:\xampp\htdocs\movies\var\cache\dev\ContainerD8KqedL\PaginatorInterface_82dac15.php:30)   at ContainerD8KqedL\PaginatorInterface_82dac15->paginate(object(Query), 1, 3)      (C:\xampp\htdocs\movies\src\Controller\MoviesController.php:52)   at App\Controller\MoviesController->index(object(Request), object(PaginatorInterface_82dac15), object(EntityManager_9a5be93))      (C:\xampp\htdocs\movies\vendor\symfony\http-kernel\HttpKernel.php:152)   at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)      (C:\xampp\htdocs\movies\vendor\symfony\http-kernel\HttpKernel.php:74)   at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)      (C:\xampp\htdocs\movies\vendor\symfony\http-kernel\Kernel.php:202)   at Symfony\Component\HttpKernel\Kernel->handle(object(Request))      (C:\xampp\htdocs\movies\vendor\symfony\runtime\Runner\Symfony\HttpKernelRunner.php:35)   at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()      (C:\xampp\htdocs\movies\vendor\autoload_runtime.php:29)   at require_once('C:\\xampp\\htdocs\\movies\\vendor\\autoload_runtime.php')      (C:\xampp\htdocs\movies\public\index.php:5)


</body>
</html>

**Comment by n3ill19** [comment]
Doctrine\ORM\Query\QueryException:
SELECT .. FROM App\Entity\Movie

  at C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\QueryException.php:25
  at Doctrine\ORM\Query\QueryException::dqlError('SELECT .. FROM App\\Entity\\Movie')
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:494)
  at Doctrine\ORM\Query\Parser->syntaxError('IdentificationVariable | ScalarExpression | AggregateExpression | FunctionDeclaration | PartialObjectExpression | "(" Subselect ")" | CaseExpression', array('value' => '.', 'type' => 10, 'position' => 7))
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:2308)
  at Doctrine\ORM\Query\Parser->SelectExpression()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:1227)
  at Doctrine\ORM\Query\Parser->SelectClause()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:914)
  at Doctrine\ORM\Query\Parser->SelectStatement()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:883)
  at Doctrine\ORM\Query\Parser->QueryLanguage()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:293)
  at Doctrine\ORM\Query\Parser->getAST()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Parser.php:393)
  at Doctrine\ORM\Query\Parser->parse()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Pagination\Paginator.php:252)
  at Doctrine\ORM\Tools\Pagination\Paginator->unbindUnusedQueryParams(object(Query))
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Pagination\Paginator.php:241)
  at Doctrine\ORM\Tools\Pagination\Paginator->getCountQuery()
     (C:\xampp\htdocs\movies\vendor\doctrine\orm\lib\Doctrine\ORM\Tools\Pagination\Paginator.php:115)
  at Doctrine\ORM\Tools\Pagination\Paginator->count()
  at count(object(Paginator))
     (C:\xampp\htdocs\movies\vendor\knplabs\knp-components\src\Knp\Component\Pager\Event\Subscriber\Paginate\Doctrine\ORM\QuerySubscriber.php:46)
  at Knp\Component\Pager\Event\Subscriber\Paginate\Doctrine\ORM\QuerySubscriber->items(object(ItemsEvent), 'knp_pager.items', object(TraceableEventDispatcher))
     (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\Debug\WrappedListener.php:111)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(ItemsEvent), 'knp_pager.items', object(TraceableEventDispatcher))
     (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'knp_pager.items', object(ItemsEvent))
     (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(ItemsEvent), 'knp_pager.items')
     (C:\xampp\htdocs\movies\vendor\symfony\event-dispatcher\Debug\TraceableEventDispatcher.php:152)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(ItemsEvent), 'knp_pager.items')
     (C:\xampp\htdocs\movies\vendor\knplabs\knp-components\src\Knp\Component\Pager\Paginator.php:87)
  at Knp\Component\Pager\Paginator->paginate(object(Query), 1, 3, array('pageParameterName' => 'page', 'sortFieldParameterName' => 'sort', 'sortDirectionParameterName' => 'direction', 'filterFieldParameterName' => 'filterField', 'filterValueParameterName' => 'filterValue', 'distinct' => true, 'pageOutOfRange' => 'ignore', 'defaultLimit' => 10))
     (C:\xampp\htdocs\movies\var\cache\dev\ContainerD8KqedL\PaginatorInterface_82dac15.php:30)
  at ContainerD8KqedL\PaginatorInterface_82dac15->paginate(object(Query), 1, 3)
     (C:\xampp\htdocs\movies\src\Controller\MoviesController.php:52)
  at App\Controller\MoviesController->index(object(Request), object(PaginatorInterface_82dac15), object(EntityManager_9a5be93))
     (C:\xampp\htdocs\movies\vendor\symfony\http-kernel\HttpKernel.php:152)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (C:\xampp\htdocs\movies\vendor\symfony\http-kernel\HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (C:\xampp\htdocs\movies\vendor\symfony\http-kernel\Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (C:\xampp\htdocs\movies\vendor\symfony\runtime\Runner\Symfony\HttpKernelRunner.php:35)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (C:\xampp\htdocs\movies\vendor\autoload_runtime.php:29)
  at require_once('C:\\xampp\\htdocs\\movies\\vendor\\autoload_runtime.php')
     (C:\xampp\htdocs\movies\public\index.php:5)

**Reply by norkunas** [reply]
So replace `..` with your alias :)

**Reply by n3ill19** [reply]
I repair - Give me a minute and I'll post it fixed

**Comment by n3ill19** [comment, marked as answer]
It's my first blog with symfony - but Thanks for help!

![done](https://user-images.githubusercontent.com/9141325/174396562-ae96e7d8-e30d-4739-819c-d9539780c16a.jpg)
